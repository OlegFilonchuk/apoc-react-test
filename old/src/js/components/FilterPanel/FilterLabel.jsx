import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './FilterLabel.less';

export default function FilterLabel({ label, labelIcon, labelCount, className }) {
  const finalClassName = classNames('label-item', className);

  const labelItemIcon = labelIcon ? (
    <span className="label-item__icon">
      <i className={labelIcon} />
    </span>
  ) : null;

  return (
    <div className={finalClassName}>
      {labelItemIcon}
      <div className="label-item__name">{label}</div>
      {labelCount !== null && <div className="label-item__number">({labelCount})</div>}
    </div>
  );
}

FilterLabel.propTypes = {
  /**
   * Label for filter item
   */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),

  /**
   * Icon for filter item
   */
  labelIcon: PropTypes.string,

  /**
   * Label for filter item
   */
  labelCount: PropTypes.node,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

FilterLabel.defaultProps = {
  label: '',
  labelIcon: '',
  labelCount: null,
  className: ''
};
