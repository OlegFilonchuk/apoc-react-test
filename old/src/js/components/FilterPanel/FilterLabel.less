.label-item {
  display: flex;

  &__icon {
    margin-right: 5px;
  }

  &__name {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__number {
    margin-left: 5px;
  }
}
