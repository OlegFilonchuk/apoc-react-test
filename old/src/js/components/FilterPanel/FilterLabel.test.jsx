import React from 'react';
import { shallow } from 'enzyme';

import FilterLabel from './FilterLabel';

describe('<FilterLabel />', () => {
  it('should render correct html', () => {
    const filterLabel = shallow(<FilterLabel label={'testLabel'} userProfilesCount={1} />);

    expect(filterLabel).toMatchSnapshot();
  });
});
