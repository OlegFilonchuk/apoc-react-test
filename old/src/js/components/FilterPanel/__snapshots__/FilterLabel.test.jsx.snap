// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterLabel /> should render correct html 1`] = `
<div
  className="label-item"
>
  <div
    className="label-item__name"
  >
    testLabel
  </div>
</div>
`;
